### PR TITLE
Add automatic formatting via prettier

### PR DESCRIPTION
**Pull request by marvinhagemeister**
Long time coming :tada: 

- [x] Add prettier
- [x] Format all source code
- [x] Apply formatting via git hooks automatically
- [x] Make sure that it has no effect on size :tada: 
- [x] Update eslint config

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,10 +36,11 @@
     "lint": "eslint src test debug compat hooks test-utils"
   },
   "eslintConfig": {
-    "extends": "developit",
+    "extends": ["developit", "prettier"],
     "settings": {
       "react": {
-        "pragma": "createElement"
+        "pragma": "createElement",
+        "version": "latest"
```
Comment (prateekbh) [inline package.json]: is this really needed in this PR?

Comment (marvinhagemeister) [reply package.json]: Alright, reverted it :+1:

File: compat/src/index.js [hunk 2]
```diff
@@ -186,8 +186,9 @@ let Children = {
 	},
 	only(children) {
 		children = toChildArray(children);
-		if (children.length !== 1)
+		if (children.length !== 1) {
```
Comment (prateekbh) [inline compat/src/index.js]: wait doesn't the git pre-commit hook not revert this change for you?

**Comment by andrewiggins** [comment]
Oh! I also confirmed on my machine that this PR doesn't change gzip size for any of our modules

File: compat/src/index.js (hunk 2 above)
Comment (ForsakenHarmony) [reply compat/src/index.js]: Seems to keep it if it's there
